File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    s=s.lower()
    S=list(s)
    # 첫문자이거나 이전에 공백이면 대문자
    # 조건 : 알파벳
    if S[0].isalpha():
        S[0]=S[0].upper()
        
    for i in range(1,len(S)):
        if S[i-1]==" " and S[i].isalpha():
            S[i]=S[i].upper()
    return "".join(S)

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
// s.split(" ",-1)을 하면 빈문자열까지 포함하라는 의미이다. 
// String_변수명.toLowerCase(), String_변수명.toUpperCase()

import java.util.*;
class Solution {
    public String solution(String s) {
        String answer = "";
        String[] sentences = s.split(" ",-1);
        List<String> candidate = new ArrayList<>();
        
        for (String sent : sentences){
            if (sent.isBlank()){
                candidate.add(sent);
                continue;
            }
            String lowerSent = sent.toLowerCase();
            candidate.add(lowerSent.substring(0,1).toUpperCase()+lowerSent.substring(1));
        }
        
        answer = String.join(" ",candidate);

        return answer;
    }
}